ambry-store/src/test/java/com.github.ambry.store/IndexTest.java

```diff
@@ -943,6 +1083,37 @@ private void verifyLogSegmentCount(int expectedCount) {
   }
 
   /**
+   * Verify that deleted entries are hard deleted or not based on the {@code hardDeleted} flag
+   * @param hardDeleted {@code true} if entries are hard deleted, {@code false} otherwise
+   * @throws IOException
+   */
+  private void verifyEntriesForHardDeletes(boolean hardDeleted) throws IOException {
+    for (MockId id : deletedKeys) {
+      IndexValue putValue = allKeys.get(id).getFirst();
+      Offset offset = putValue.getOffset();
+      LogSegment segment = log.getSegment(offset.getName());
+      long size = putValue.getSize() - HARD_DELETE_START_OFFSET - HARD_DELETE_LAST_PART_SIZE;
+      ByteBuffer readBuf = ByteBuffer.allocate((int) size);
+      segment.readInto(readBuf, offset.getOffset() + HARD_DELETE_START_OFFSET);
+      readBuf.flip();
+      if (hardDeleted) {
+        while (readBuf.hasRemaining()) {
+          assertEquals("Hard delete has not zeroed out the data", (byte) 0, readBuf.get());
+        }
+      } else {
+        boolean foundOnes = false;
+        while (readBuf.hasRemaining()) {
+          if (Byte.compare(readBuf.get(), (byte) 1) == 0) {
+            foundOnes = true;
```

Comment:
how do you know the data has at least one byte that is exactly equal to 1? It is random data, is it not?